cmd/dagger/module.go

```diff
@@ -94,30 +94,41 @@ func getCompatVersion() string {
 	return compatVersion
 }
 
-func init() {
-	moduleFlags.StringVarP(&moduleURL, "mod", "m", "", "Path to the module directory. Either local path or a remote git repo")
+func moduleAddFlags(cmd *cobra.Command, persistent bool) {
+	var flags *pflag.FlagSet
+	if persistent {
+		flags = cmd.PersistentFlags()
+	} else {
+		flags = cmd.Flags()
+	}
+	flags.StringVarP(&moduleURL, "mod", "m", "", "Module reference to load, either a local path or a remote git repo (defaults to current directory)")
+	flags.BoolVarP(&moduleNoURL, "no-mod", "M", false, "Don't automatically load a module (mutually exclusive with --mod)")
+	cmd.MarkFlagsMutuallyExclusive("mod", "no-mod")
+
 	var defaultAllowLLM []string
 	if allowLLMEnv := os.Getenv("DAGGER_ALLOW_LLM"); allowLLMEnv != "" {
 		defaultAllowLLM = strings.Split(allowLLMEnv, ",")
 	}
-	moduleFlags.StringSliceVar(&allowedLLMModules, "allow-llm", defaultAllowLLM, "List of URLs of remote modules allowed to access LLM APIs, or 'all' to bypass restrictions for the entire session")
+	flags.StringSliceVar(&allowedLLMModules, "allow-llm", defaultAllowLLM, "List of URLs of remote modules allowed to access LLM APIs, or 'all' to bypass restrictions for the entire session")
+}
 
+func init() {
 	for _, fc := range funcCmds {
 		if !fc.DisableModuleLoad {
-			fc.Command().PersistentFlags().AddFlagSet(moduleFlags)
+			moduleAddFlags(fc.Command(), true)
 		}
 	}
 
-	funcListCmd.PersistentFlags().AddFlagSet(moduleFlags)
-	listenCmd.PersistentFlags().AddFlagSet(moduleFlags)
-	queryCmd.PersistentFlags().AddFlagSet(moduleFlags)
-	configCmd.PersistentFlags().AddFlagSet(moduleFlags)
+	moduleAddFlags(funcListCmd, true)
```

Comment:
`dagger functions` could at some point return core functions if `-M` is used, but I don't see a point in allowing it if it's just going to return an error.

Reply:
In other words, if a user calls `dagger functions --help`, I think it's reasonable to expect that what's listed as available options can in fact be used and not error without a solution.